### PR TITLE
Fix broken links on summary page

### PR DESCRIPTION
**Pull request by renaynay**

### CODE REVIEW

**Comment by adlerjohn** [comment]
Will take a look tmr. Relevant: https://github.com/rust-lang/mdBook/issues/984

File: src/README.md [hunk 1]
```diff
@@ -1,14 +1,14 @@
 # Celestia Specifications
 
-- [Specification](./specs/README.md)
+- [Specification](specs/specification.md)
```
Comment (adlerjohn) [inline src/README.md:3]: For uniformity, make links start with `./`. That's about it.

**Review by adlerjohn — COMMENTED** [review]
It looks like there's an outstanding bug in mdBook, so your workaround is the only solution until that's fixed. Left a minor change request.

File: src/README.md (hunk 1 above)
Comment (liamsi) [reply src/README.md:3]:
```suggestion
- [Specification](./specs/specification.md)
```

File: src/README.md [hunk 2]
```diff
@@ -1,14 +1,14 @@
...
+- [Specification](./specs/specification.md)
   - [Architecture](./specs/architecture.md)
   - [Data Structures](./specs/data_structures.md)
   - [Consensus](./specs/consensus.md)
   - [Block Proposer](./specs/block_proposer.md)
   - [Networking](./specs/networking.md)
   - [Light client](./specs/light_client.md)
   - [Node types](./specs/node_types.md)
-- [Rationale](./rationale/README.md)
+- [Rationale](rationale/rationale.md)
```
Comment (adlerjohn) [inline src/README.md:11]:
```suggestion
- [Rationale](./rationale/rationale.md)
```

File: src/SUMMARY.md [hunk 3]
```diff
@@ -2,15 +2,15 @@
 
 [Celestia Specifications](./README.md)
 
-- [Specification](./specs/README.md)
+- [Specification](specs/specification.md)
```
Comment (adlerjohn) [inline src/SUMMARY.md:5]:
```suggestion
- [Specification](./specs/specification.md)
```

File: src/SUMMARY.md [hunk 4]
```diff
@@ -2,15 +2,15 @@
...
   - [Architecture](./specs/architecture.md)
   - [Data Structures](./specs/data_structures.md)
   - [Consensus](./specs/consensus.md)
   - [Block Proposer](./specs/block_proposer.md)
   - [Networking](./specs/networking.md)
   - [Light client](./specs/light_client.md)
   - [Node types](./specs/node_types.md)
-- [Rationale](./rationale/README.md)
+- [Rationale](rationale/rationale.md)
```
Comment (adlerjohn) [inline src/SUMMARY.md:13]:
```suggestion
- [Rationale](./rationale/rationale.md)
```

**Review by liamsi — APPROVED** [review]
Thanks Rene!

**Review by adlerjohn — APPROVED** [review]

**Pull request merged**